website/src/_posts/2020-03-06-python-typing.md

```diff
@@ -0,0 +1,117 @@
+---
+layout: post
+title:  "Python SDK Typing Changes"
+date:   2020-03-06 00:00:01 -0800
+excerpt_separator: <!--more-->
+categories: blog python typing
+authors:
+  - chadrik
+  - udim
+
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+TODO excerpt
+
+<!--more-->
+
+Python supports type annotations on functions (PEP 484). Static type checkers,
+such as mypy, are used to verify adherence to these types.
+For example:
+```py
+def f(v: int) -> int:
+  return v[0]
+```
+Running mypy on the above code will give the error:
+`Value of type "int" is not indexable`.
+
+We've recently made changes to Beam in 2 areas:
+
+Adding type hints throughout Beam. TODO expand
+
+Second, we've added support for Python 3 type annotations. This allows SDK
+users to specify a DoFn's type hints in one place. 
+We've also expanded Beam's support of `typing` module types.
+
+For more background see: 
+[Ensuring Python Type Safety](https://beam.apache.org/documentation/sdks/python-type-safety/).
+
+# Beam Is Typed
+
+TODO
```

Comment:
@chadrik LMK if you want to write about the new type hints, mypy checking, other stuff you've added related to types.

Reply:
I definitely do, wont' have time for at least half a week.  Can you give me a timeline?  
I am currently in save-the-company mode, so I don't have much time on my hands until that's done.  Our industry is a lot like a factory.  Many humans on high-end graphics workstations operating on petabytes of data in a highly collaborative assembly line.  It's not something that's easily convertable to WFH, but we're almost there.

Reply:
The deadline is by the time 2.21 is released. That release is scheduled to be cut on Apr 8, and I estimate releasing it will take 2-6 weeks. So let's aim for Apr 22.

Reply:
So, no immediate rush. We are also adjusting to WFH

Reply:
Hi, I realize I missed our April 22nd deadline.  I can commit to getting this done by EOW.  Does that work for you?

Reply:
Yes, that works